### ISSUE TITLE
Linear transformation of measurement results

### ISSUE THREAD

**Issue by galexv** [ALPSCore/ALPSCore#305]
Request from Markus @mwallerb: it would be good to have a linear transformation of a vector `Result`, something along these lines:
```c++
namespace aa=alps::accumulators;
typedef std::vector<double> vector;
//.....
// Prepare accumulator
aa::accumulator_set aset;
aset << aa::FullBinningAccumulator<vector>("vector5");
// Fill it with data
for (int i=0; i<100; ++i) {
    aset["vector5"] << vector(5, double(i));
}
// Get the results set
aa:result_set rset(aset);

// what could be a good matrix type? 
// boost::multi_array? Eigen matrix? Any type with a subset of Eigen accessors?
some_matrix_type trans3x5=get_me_my_matrix_of_size(3,5); 

// what would be a good name for the function?
aa::result_wrapper res3=transform(trans3x5, rset["vector5"]);

assert( res3.mean<vector>().size() == 3 );
```
Markus, any additions/corrections are welcome.

**Comment by egull** [comment]
@mwallerb, is this addressed in alea?

**Comment by mwallerb** [comment]
Fixed by: #412
